telemetry/packages/e2e-tests/lib/mqtt.ts

```diff
@@ -0,0 +1,38 @@
+import mqtt from 'mqtt';
+
+export const client = mqtt.connect('mqtt://localhost:1883');
+
+type MqttMessageValidation = (receivedTopic: string, message: Buffer) => void;
+
+/**
+ * Validates a message received on an MQTT topic.
+ * @param topic The topic the message will be send on
+ * @param validate A validation function which will be called with the received topic and message values
+ * @param timeout Time to wait (in ms) before giving up
+ */
+export async function validateMqttMessage(
+  trigger: () => void,
+  validate: MqttMessageValidation,
+  timeout = 1000,
+): Promise<void> {
+  return new Promise(async (resolve, reject) => {
+    const client = mqtt.connect('mqtt://localhost:1883');
```

Comment:
Is there a specific need for renewing the MQTT client each time this validation function is run, or would a persistent client initiated in a config file perhaps be a viable alternative?

Reply:
I think I tried this and it didn't work properly. We can try this again later on if it becomes an issue :)